server/dash/pages/nc_summary_comparison.py

```diff
@@ -104,11 +358,325 @@ def generate_dynamic_filter(selected_nc):
+                             title="Share of each Request Type")
+    req_type_pie_chart.update_layout(margin=dict(l=50, r=50, b=50, t=50),
+                                  legend_title=dict(font=dict(size=10)), font=dict(size=9))
+    return req_type_pie_chart
+
+
+@callback(
+    Output("time_close_histogram", "figure"),
+    Output("data_quality_output", "children"),
+    Input("selected_nc", "value"),
+    [Input("selected_request_types", "value")],
+    Input("data_quality_switch", "value")
+)
+def generate_time_to_close_histogram(selected_nc, selected_request_types=None,
+data_quality_switch=True):
+    """Generates the request type pie chart based on selected filters.
+    This callback function takes the selected neighborhood council (nc) 
+    value from the "selected_nc" dropdown, selected request type from "selected_request_types"
+    dropdown to output a histogram for the time it takes for each request to close.
+    Args:
+        selected_nc: A string argument automatically detected by Dash callback 
+            function when "selected_nc" element is selected in the layout.
+            selected_request_types: A list of strings automatically detected by Dash 
+            callback function when "selected_request_types" element is selected in the
+            layout, default None.
+        data_quality_switch: A boolean for data quality filter automatically detected 
+            by Dash callback function when "data_quality_switch" element is selected in 
+            the layout, default True.
+    Returns:
+        time_close_histogram: histogram showing the distribution of the request time to close.
+        data_quality_output: A string stating the status of the data quality filter 
+            ("Quality Filter: On" or "Quality Filter: Off").
+    """
+    df = generate_filtered_dataframe(api_data_df, selected_nc, selected_request_types)
+    df, num_bins, data_quality_output = filter_bad_quality_data(df, data_quality_switch)
+
+    # Distribution for the total number of requests.
+    print(" * Generating requests time to close histogram")
+    time_close_histogram = px.histogram(df, x="timeToClose",
+    title="Distribution of Time to Close Request",
+     nbins=num_bins, range_x=[min(
+         df.loc[:, "timeToClose"]), max(df.loc[:, "timeToClose"])],
+        labels={"timeToClose": "Request Duration",
+         "count": "Frequency"})
+    time_close_histogram.update_layout(margin=dict(l=50, r=50, b=50, t=50), font=dict(size=9))
+    return time_close_histogram, data_quality_output
+
+@callback(
+    Output("total_req_card", "children"),
+    Output("total_req_card2", "children"),
+    Output("num_days_card", "children"),
+    Output("num_days_card2", "children"),
+    Input("nc_comp_dropdown", "value"),
+    Input("nc_comp_dropdown2", "value"),
+    prevent_initial_call=True
+)
+def generate_indicator_visuals(nc_comp_dropdown, nc_comp_dropdown2):
```

Comment:
I feel like we should have two callbacks, one for A and one for B. And each callback can call a single helper function that performs the logic.

Reply:
Thanks for the suggestion - resolved.